torch/nn/parallel/distributed.py

```diff
@@ -396,7 +401,7 @@ def __init__(
         ).format(distinct_device_types)
         self.device_type = list(distinct_device_types)[0]
 
-        if self.device_type == "cpu" or self.is_multi_device_module:
+        if device_ids is None or self.device_type == "cpu" or self.is_multi_device_module:
```

Comment:
Sorry I missed the case about backward compatibility for empty list, lets try to keep backward compatibility and add a condition here for empty list as well. Let's also add a unit test with empty list to ensure this contract is preserved. 

Its fine not to mention this in the docs though since its better that the user always passes in None.

Reply:
Added a condition here as well as a unit test for SPSD mode.